src/controller/auth.py

```diff
@@ -88,6 +93,13 @@ def google_error(google, error, error_description=None, error_uri=None):
     )
     abort(503, message)
 
+@auth.route('/login', methods=['GET'])
+def login():
+    redirect_url = request.args.get('next')
```

Comment:
The 'next' argument will not be passed in as a query parameter but instead already set in the session as `next` as configured currently. All you need here is to redirect and then read from `next` in the callback. If `next` is overriden by the time the callback is executed, then we should copy over the session data from `next` into `redirect_url` in `/login` and then read from there in the callback.

Reply:
fixed